src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java

```diff
@@ -72,4 +72,11 @@ default LearningPath findWithEagerCompetenciesAndLearningObjectsByIdElseThrow(lo
     default LearningPath findWithEagerCompetenciesAndLearningObjectsAndCompletedUsersByIdElseThrow(long learningPathId) {
         return findWithEagerCompetenciesAndLearningObjectsAndCompletedUsersById(learningPathId).orElseThrow(() -> new EntityNotFoundException("LearningPath", learningPathId));
     }
+
+    @Query("""
+            SELECT COUNT (learningPath)
+            FROM LearningPath learningPath
+            WHERE learningPath.course.id = :#{#courseId}
+            """)
+    long countByCourseId(@Param("courseId") long courseId);
```

Comment:
I think you don't even need the Query here, JPA should be able to parse the name itself

Reply:
This query was actually the reason for the stuck health status warning. This is now resolved 👍